cordova-lib/src/cordova/restore-util.js

```diff
@@ -63,7 +64,27 @@ function installPlatformsFromConfigXML(platforms, opts) {
             // Combining arrays and checking duplicates.
             comboArray = pkgJsonPlatforms.slice();
         }
+
         engines = cfg.getEngines(projectHome);
+
+        if(fs.existsSync(pkgJsonPath)) {
+            pkgJson = require(pkgJsonPath);
+
+        // If package.json doesn't exist, auto-create one.
+        } else if (engines.length > 0 && !fs.existsSync(pkgJsonPath)) {
```

Comment:
It would be better to change this from `if (engines.length > 0 && !fs.existsSync(pkgJsonPath))` to `if (engines.length > 0 && pkgJson))` Since you already check for `pkgJsonPath` and `require pkgjson` on lines 53-54, you don't have to do another `existsSync` operation (which is expensive).